pkg/correlation/context.go

```diff
@@ -61,3 +65,10 @@ func WithCorrelationContext(ctx context.Context, origin Component) context.Conte
 
 	return ctx
 }
+
+// TODO is an alias for context.TODO(), specifically
+// for keeping track of areas where we might want to add
+// the correlation context.
+func TODO() context.Context {
```

Comment:
nit: Why have a TODO function in correlation package. Why not use context.TODO() where you have correlation.TODO()?

Reply:
Good question, we have this to mark spots where we MAY want to add the correlation context in the future. context.TODO() is nice for a general TODO, where correlation.TODO() is for us to easily search the whole codebase for spots to add correlation context.

Reply:
Others writing code in porx may use context.TODO() which would make it harder to remember where we may want to add the correlation context.

Reply:
In that case would it be better with something like a correlation.GetContext()? Then you need to change only the implementation?

Reply:
Not necessarily, context.Context objects are typically generated once at the beginning of a call stack and passed down. So it's more likely that it'll be added as a function parameter and we'll need to replace `context.TODO()` with the context object parameter.

I.e.
```go
func someFunc() {
  logrus.WithContext(correlation.TODO()).Infof("...")
}
```

refactored to 
```go
func someFunc(ctx context.Context) {
  logrus.WithContext(ctx).Infof("...")
}
```

Reply:
Makes sense. Thanks.